genesyscloud/routing_utilization_label/data_source_genesyscloud_routing_utilization_label.go

```diff
@@ -0,0 +1,33 @@
+package routing_utilization_label
+
+import (
+	"context"
+	"fmt"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceRoutingUtilizationLabelRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	sdkConfig := m.(*provider.ProviderMeta).ClientConfig
+	proxy := getRoutingUtilizationProxy(sdkConfig)
+	name := d.Get("name").(string)
+
+	return util.WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
+		label, resp, getErr := proxy.getRoutingUtilizationLabelByName(ctx, name)
+
+		if getErr != nil {
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Error requesting label %s | error: %s", name, getErr), resp))
+		}
+
+		if label == nil {
+			return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("No labels found with name %s", name), resp))
+		}
+		d.SetId(*label.Id)
```

Comment:
I had a look at the proxy function `getRoutingUtilizationLabelByName`, and it will return an error if no label was found, so this code that checks `if label == nil` won't be reached. It would probably be best to return a bool to know if its retryable